spec/es_spec_helper.rb

```diff
@@ -20,7 +20,9 @@ def get_host_port
   end
 
   def get_client
-    Elasticsearch::Client.new(:hosts => [get_host_port])
+    client = Elasticsearch::Client.new(:hosts => [get_host_port])
+    client.instance_variable_set("@verified", true) # bypass client side version checking
+    client
```

Comment:
While either approach is technically messing with Elasticsearch's private internals, I greatly prefer mocking methods over directly manipulating the ivar state of an object we don't own.

Both are liable to create problems down the track if the Elasticsearch gem decides to change their internals, but with a method mock we have both a greater sense of the _intent_ behind the patch and a somewhat smaller likelihood of creating problems (e.g., if the ivar takes on a different semantic meaning we risk stomping on it with something invalid; while if the method signature changes we risk not intercepting).

```suggestion
    Elasticsearch::Client.new(:hosts => [get_host_port]).tap do |client|
      allow(client).to receive(:verify_elasticsearch).and_return(true)
    end
```

Reply:
Thanks for the suggestions. The client team plans to release a version that relaxes the checking in 7.16. This hack is for a short-term transition. But I think it is no harm to take your suggestion too